.semaphore/semaphore.yml

```diff
@@ -11,52 +21,116 @@ global_job_config:
+  - name: 'Build / Cache arm64 and ppc64le'
+    dependencies: [ "Build amd64" ]
+    task:
       prologue:
         commands:
-          - checkout
-          - >-
-            if [[ -v SEMAPHORE_GIT_PR_NUMBER ]]; then
-              unset QUAY_USERNAME QUAY_TOKEN; docker logout; docker logout quay.io;
-            else
-              docker login -u="$QUAY_USERNAME" -p="$QUAY_TOKEN" quay.io;
-            fi
-          - cache restore go-pkg-cache
-          - cache restore go-mod-cache
-      epilogue:
-        on_pass:
+          - cache restore go-pkg-cache-amd64-${SEMAPHORE_GIT_SHA}
+          - cache restore go-mod-cache-amd64-${SEMAPHORE_GIT_SHA}
+      jobs:
+        - name: Build arm64
+          execution_time_limit:
+            minutes: 45
+          commands:
+            - make register build ARCH=arm64
+            # Store in cache for other builds in this run
+            - 'cache store bin-arm64-${SEMAPHORE_GIT_SHA} build/_output/bin'
+            - 'cache store go-pkg-cache-arm64-${SEMAPHORE_GIT_SHA} .go-pkg-cache'
+            - 'cache store go-mod-cache-arm64-${SEMAPHORE_GIT_SHA} ${HOME}/go/pkg/mod/cache'
+        - name: Build ppc64le
+          execution_time_limit:
+            minutes: 45
+          commands:
+            - make register build ARCH=ppc64le
+            # Store in cache for other builds in this run
+            - 'cache store bin-ppc64le-${SEMAPHORE_GIT_SHA} build/_output/bin'
+            - 'cache store go-pkg-cache-ppc64le-${SEMAPHORE_GIT_SHA} .go-pkg-cache'
+            - 'cache store go-mod-cache-ppc64le-${SEMAPHORE_GIT_SHA} ${HOME}/go/pkg/mod/cache'
+  - name: 'Static / File Generation Checks'
+    dependencies: [ "Build amd64" ]
+    task:
+      jobs:
+        - name: Static / File Generation Checks, Vet
+          execution_time_limit:
+            minutes: 15
           commands:
-            - cache store go-pkg-cache .go-pkg-cache
-            - 'cache store go-mod-cache ${HOME}/go/pkg/mod/cache'
+            - make format-check validate-gen-versions fmt
+            - make dirty-check
+            - make vet
+
+  - name: 'UT'
+    dependencies: [ "Build amd64" ]
+    task:
+      jobs:
+        - name: Run UTs
+          execution_time_limit:
+            minutes: 30
+          commands:
+            - make ut
+            - make dirty-check
+
+  - name: 'FV'
+    dependencies: [ "Build amd64" ]
+    task:
+      agent:
+        machine:
+          type: e1-standard-4
+          os_image: ubuntu1804
+      jobs:
+        - name: Run FVs
+          execution_time_limit:
+            minutes: 30
+          commands:
+            - make fv
+            - make dirty-check
+
+promotions:
+  - name: Push Images
+    pipeline_file: push_images.yml
+    auto_promote:
+      when: "branch =~ 'master|release-.*'"
+  - name: Clean Up
+    pipeline_file: clean_up.yml
+    # Don't auto promote if this is master or a release branch so the cache is available for the Push Images pipeline.
+#    auto_promote:
```

Comment:
Should this auto_promote when something fails? If that happens then don't we need to rerun anyway so there won't be any use of the cache and ideally the failure will mean we need to make an update which would mean the current cache won't be used either.

Reply:
Probably should promote when it fails or not on master or release branch.

Reply:
This was just temporarily commented out, it will auto promote for non master / release branches. The push images pipeline is responsible for promoting to clean on master / release branches (if the pipeline doesn't fail, there's a comment there about that).

Reply:
Because we'd likely want to either 1. rerun the job or 2. log into the box and try cutting the images manually. If we promote on failure then the go package cache would be clear, making either of those options take 20 minutes instead of 2.

So I see it as quite important to not clear the cache unless it passes.

Reply:
Those seem like edge cases where more generally I think clearing the cache is the right thing to do. I guess if we find not clearing the cache is problematic we can change it then.